bestie/utils/googlebench/bestie_reporter_test.cc

```diff
@@ -0,0 +1,735 @@
+              testing::ContainsRegex("bestie/utils/googlebench.*/test.metrics.pb"));
+  EXPECT_THAT(bestie::MetricsPath(1),
+              testing::ContainsRegex("bestie/utils/googlebench.*/test001.metrics.pb"));
+  EXPECT_THAT(bestie::MetricsPath(123, "freedom"),
+              testing::ContainsRegex("bestie/utils/googlebench.*/freedom123.metrics.pb"));
+}
+
+const std::string kExpectedMessageText = R"(
+metrics {
+  metricname: "context__cpu_info__num_cpus"
+  tags {
+    key: "context__sys_info__name"
+    value: "<hostname>"
+  }
+  tags {
+    key: "context__executable_name"
+    value: "<path-of-binary>"
+  }
+}
+metrics {
+  metricname: "context__cpu_info__cycles_per_second"
+  tags {
+    key: "context__sys_info__name"
+    value: "<hostname>"
+  }
+  tags {
+    key: "context__executable_name"
+    value: "<path-of-binary>"
+  }
+}
+metrics {
+  metricname: "context__cpu_info__load_avg__0"
+  tags {
+    key: "context__sys_info__name"
+    value: "<hostname>"
+  }
+  tags {
+    key: "context__executable_name"
+    value: "<path-of-binary>"
+  }
+}
+metrics {
+  metricname: "context__cpu_info__load_avg__1"
+  tags {
+    key: "context__sys_info__name"
+    value: "<hostname>"
+  }
+  tags {
+    key: "context__executable_name"
+    value: "<path-of-binary>"
+  }
+}
+metrics {
+  metricname: "context__cpu_info__load_avg__2"
+  tags {
+    key: "context__sys_info__name"
+    value: "<hostname>"
+  }
+  tags {
+    key: "context__executable_name"
+    value: "<path-of-binary>"
+  }
+}
+metrics {
+  metricname: "run__iterations"
+  tags {
+    key: "run__benchmark_name"
+    value: "Bench"
+  }
+  tags {
+    key: "context__sys_info__name"
+    value: "<hostname>"
+  }
+  tags {
+    key: "context__executable_name"
+    value: "<path-of-binary>"
+  }
+}
+metrics {
+  metricname: "run__cpu_accumulated_time"
+  tags {
+    key: "run__benchmark_name"
+    value: "Bench"
+  }
+  tags {
+    key: "context__sys_info__name"
+    value: "<hostname>"
+  }
+  tags {
+    key: "context__executable_name"
+    value: "<path-of-binary>"
+  }
+}
+metrics {
+  metricname: "run__real_accumulated_time"
+  tags {
+    key: "run__benchmark_name"
+    value: "Bench"
+  }
+  tags {
+    key: "context__sys_info__name"
+    value: "<hostname>"
+  }
+  tags {
+    key: "context__executable_name"
+    value: "<path-of-binary>"
+  }
+}
+metrics {
+  metricname: "run__adjusted_cpu_time"
+  tags {
+    key: "run__benchmark_name"
+    value: "Bench"
+  }
+  tags {
+    key: "context__sys_info__name"
+    value: "<hostname>"
+  }
+  tags {
+    key: "context__executable_name"
+    value: "<path-of-binary>"
+  }
+}
+metrics {
+  metricname: "run__adjusted_real_time"
+  tags {
+    key: "run__benchmark_name"
+    value: "Bench"
+  }
+  tags {
+    key: "context__sys_info__name"
+    value: "<hostname>"
+  }
+  tags {
+    key: "context__executable_name"
+    value: "<path-of-binary>"
+  }
+}
+metrics {
+  metricname: "run__maxheapbytes_used"
```

Comment:
Out of curiosity, why is this block replicated twice in kExpectedMessageText:

```
metrics {
  metricname: "run__maxheapbytes_used"
  tags {
    key: "run__benchmark_name"
    value: "Message"
  }
  tags {
    key: "run__skipped"
    value: "message"
  }
  tags {
    key: "run__skip_message"
    value: "For the powerful, crimes are those that others commit."
  }
  tags {
    key: "context__sys_info__name"
    value: "<hostname>"
  }
  tags {
    key: "context__executable_name"
    value: "<path-of-binary>"
  }
}
```

Reply:
good catch, cleaned.